### PR TITLE
hub: fix task execution with rootless podman

### PR DESCRIPTION
**Pull request by lzaoral**
Depends on #67!

Resolves: https://gitlab.cee.redhat.com/covscan/covscan/-/issues/255

### CODE REVIEW

**Comment by lzaoral** [comment]
Blocks #68.

**Comment by lzaoral** [comment]
@siteshwar Do we want to halt the testing farm job when there are merge conflicts with `main`?  I'm a bit surprised that this is the default behaviour, to be honest.

**Comment by lzaoral** [comment]
This Packit setting might be related: https://packit.dev/docs/configuration#merge_pr_in_ci

**Comment by siteshwar** [comment]
> This Packit setting might be related: https://packit.dev/docs/configuration#merge_pr_in_ci

Feel free to disable this setting in our project.

**Comment by lzaoral** [comment]
This PR fixes all known issues with rootless Podman on Linux and macOS.  The integration test-suite should now execute properly on any x86_64 platform when Docker or Podman are in use.

The integration test-suite still runs only partially on `aarch64` because the `Fedora-37` tag is hard-wired to the `x86_64` buildroot.  If you want to fully execute this script, you need to *manually* change the buildroot in the admin interface right after the containers are started.

**Comment by lzaoral** [comment]
@jperezdealgaba Could you please test the integration test suite with Docker on macOS?  Note that there are some manual steps you need to take on `aarch64` as noted here: https://github.com/openscanhub/openscanhub/pull/74#issuecomment-1673218028

**Comment by lzaoral** [comment]
@jperezdealgaba Cool!  In that case, could you, as a reviewer, approve this PR?

**Review by jperezdealgaba — APPROVED** [review]
Tested and approved

**Comment by lzaoral** [comment]
@siteshwar Can you take a look, please?  It is rather inconvenient to have all these changes stashed so that I can work on other PRs that require a functional worker.

File: containers/scripts/generate_integration_test_coverage.sh [hunk 1]
```diff
@@ -30,16 +30,21 @@ main() {
 
     ./containers/scripts/init-db.sh --full-dev --minimal "$FORCE"
 
+    # The container may have a different arch than the host!  Moreover, uname -m
+    # on macOS on Apple Silicon reports arm64 which is not recognised by Fedora.
+    ARCH=$(podman exec osh-client uname -m)
```
Comment (siteshwar) [inline containers/scripts/generate_integration_test_coverage.sh:35]: Have you tested this command on Fedora? It was leaving a trailing carriage return on my system, which was causing `osh-cli` command to fail.

Comment (siteshwar) [reply containers/scripts/generate_integration_test_coverage.sh:35]: I tried it on Fedora and it works. The trailing carriage return only appears when you run the command interactively.

**Review by siteshwar — APPROVED** [review]

File: containers/scripts/generate_integration_test_coverage.sh (hunk 1 above)
Comment (lzaoral) [reply containers/scripts/generate_integration_test_coverage.sh:35]: Oh, this is interesting:
```console
$ podman run centos:stream8 uname -m | hexdump -C
00000000  61 61 72 63 68 36 34 0a                           |aarch64.|
00000008

$ podman run -it centos:stream8 uname -m | hexdump -C
00000000  61 61 72 63 68 36 34 0d  0a                       |aarch64..|
00000009
```
Sice this script is not using `-it` the returned string does not end with `\r\n` so it should be fine.

**Pull request merged**